pkg/deploy/lattice/access_log_subscription_manager_test.go

```diff
@@ -27,19 +29,37 @@ const (
 	accessLogSubscriptionId  = "als-12345678901234567"
 )
 
-func Test_Create_NewAccessLogSubscriptionForServiceNetwork_ReturnsSuccess(t *testing.T) {
+var accessLogPolicyNamespacedName = types.NamespacedName{
+	Namespace: "test-namespace",
+	Name:      "test-name",
+}
+
+func setup(t *testing.T) (
+	context.Context,
+	*services.MockLattice,
+	an_aws.Cloud,
+	services.Tags,
+) {
 	c := gomock.NewController(t)
 	defer c.Finish()
```

Comment:
curious how test work since `defer c.Finish()` will be called after setup return but before test case starts

Reply:
Also you dont need setup func, if you follow previously proposed approach with subtest. Here is [example](https://github.com/aws/aws-application-networking-k8s/blob/main/pkg/deploy/lattice/service_manager_test.go#L18-L31)

```go
func TestFoo (t *testing.T) {
  c := gomock.NewController(t)
  defer c.Finish()
  mockLattice := mocks.NewMockLattice(c)
       
  t.Run("Create", func(t *testing.T) {
    mockLattice.EXPECT()...
  })

    t.Run("Delete", func(t *testing.T) {
    mockLattice.EXPECT()...
  })
}

```

Reply:
Fixed next commit!